gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/StructuredQuery.java

```diff
@@ -103,13 +102,17 @@
     Filter() {
     }
 
-    protected abstract DatastoreV1.Filter toPb();
+    protected abstract com.google.datastore.v1beta3.Filter toPb();
 
-    static Filter fromPb(DatastoreV1.Filter filterPb) {
-      if (filterPb.hasCompositeFilter()) {
-        return CompositeFilter.fromPb(filterPb.getCompositeFilter());
+    static Filter fromPb(com.google.datastore.v1beta3.Filter filterPb) {
+      switch (filterPb.getFilterTypeCase()) {
+        case COMPOSITE_FILTER:
+          return CompositeFilter.fromPb(filterPb.getCompositeFilter());
+        case PROPERTY_FILTER:
+          return PropertyFilter.fromPb(filterPb.getPropertyFilter());
+        default:
```

Comment:
Similar to the comment in `GqlQuery.java`, filterPb's type could potentially be unset, and I handle it the same was as described in the other comment.  Any thoughts on whether there's a cleaner way to do this?